libethereum/BlockQueue.h

```diff
@@ -67,6 +67,117 @@ enum class QueueStatus
 
 std::ostream& operator<< (std::ostream& os, QueueStatus const& obj);
 
+template<class T>
+class SizedBlockQueue
+{
+public:
+	std::size_t count() const
+	{
+		return m_queue.size();
+	}
+
+	std::size_t size() const
+	{
+		return m_size;
+	}
+	
+	bool isEmpty() const
+	{
+		return m_queue.empty();
+	}
+
+	h256 nextHash() const
+	{
+		return m_queue.front().verified.info.sha3Uncles();
+	}
+
+	T const& next() const
+	{
+		return m_queue.front();
+	}
+
+	template<class Op>
+	void forEach(Op _op) const
```

Comment:
That, I guess, means just implementing begin() & end() ?
(never saw yet boost range algorithms used in our code)
Yeah, that would be more idiomatic, I'll try it.